arm/Microsoft.ManagedIdentity/userAssignedIdentities/.bicep/nested_rbac.bicep

```diff
@@ -0,0 +1,12 @@
+param roleAssignment object
+param builtInRoleNames object
+param userMsiName string
+
+resource nested_rbac 'Microsoft.ManagedIdentity/userAssignedIdentities/providers/roleAssignments@2020-04-01-preview' = [for i in range(0, length(roleAssignment.principalIds)): {
```

Comment:
Please align with example [here](https://github.com/Azure/ResourceModules/blob/f5e8f351b69ec86ef33aa4a24daeb1f71c498e54/arm/Microsoft.Automation/automationAccounts/.bicep/nested_rbac.bicep). 
In the back we're aligning all modules to use this schema. (=> paramName, foreach-loop, name)

Reply:
@MrMCake that nested rbac deploys at the resource group scope. I can't see the scope being the Azure resource. Am I missing something here? to my knowledge, we either need to use the scope: property, or deploy it as a child like this

`managedIdentityNameMicrosoft.Authorization/roleAssignmentGUID`

Reply:
It should be:

```json
param roleAssignment object
param builtInRoleNames object
param resourceName string

resource nested_rbac 'Microsoft.Storage/storageAccounts/providers/roleAssignments@2020-04-01-preview' = [for principalId in roleAssignment.principalIds: {
  name: '${resourceName}/Microsoft.Authorization/${guid(resourceName, principalId, roleAssignment.roleDefinitionIdOrName)}'
  properties: {
    roleDefinitionId: (contains(builtInRoleNames, roleAssignment.roleDefinitionIdOrName) ? builtInRoleNames[roleAssignment.roleDefinitionIdOrName] : roleAssignment.roleDefinitionIdOrName)
    principalId: principalId
  }
  dependsOn: []
}]
```